frontend/src/services/QueryServiceImpl.ts

```diff
@@ -1,94 +1,27 @@
-import { injectable } from 'inversify';
+import { inject, injectable } from 'inversify';
 import 'reflect-metadata';
 import { LimitQuery } from '../shared/queries/LimitQuery';
 import { QueryResult } from '../shared/queries/QueryResult';
-import CancellationError from '../utils/CancellationError';
 import { CancellationToken } from '../utils/CancellationToken';
-import nop from '../utils/nop';
+import HTTPHelper from './HTTPHelper';
 import QueryService from './QueryService';
-import { QueryServiceOptions } from './QueryServiceOptions';
-
-/**
- * Builds the query service options.
- * @param options The partial options that very specified externally.
- * @returns The built options.
- */
-function buildOptions(
-  options: Partial<QueryServiceOptions>
-): QueryServiceOptions {
-  let { backendBaseUri } = options;
-
-  // As per the spec of QueryServiceOptions, we use the base URI of the source the frontend
-  // was loaded from, if no backend base uri was specified in the options.
-  if (!backendBaseUri) {
-    const { location } = window;
-    backendBaseUri = `${location.protocol}//${location.host}/${
-      location.pathname.split('/')[1]
-    }`;
-  }
-
-  return { backendBaseUri };
-}
 
 /**
  * The implementation of query service that performs query requests
  * via the backend.
  */
 @injectable()
 export default class QueryServiceImpl extends QueryService {
-  private readonly options: QueryServiceOptions;
-
-  public constructor(options: Partial<QueryServiceOptions> = {}) {
-    super();
-
-    this.options = buildOptions(options);
-  }
+  @inject(HTTPHelper)
+  // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+  private readonly http: HTTPHelper = null!;
```

Comment:
```suggestion
  private readonly http: HTTPHelper | null = null;
```

Reply:
It is actually guaranteed, that this is non-null, as this gets injected by DI. Sadly I couldn't get constructor injection to work, but property injection does its job here.